blazar/plugins/floatingips/floatingip_plugin.py

```diff
@@ -447,3 +455,109 @@ def query_fip_allocations(self, fips, detail=None, lease_id=None,
                         if k != 'floatingip_ids'})
 
         return fip_allocations
+
+
+class FloatingIpMonitorPlugin(monitor.GeneralMonitorPlugin, neutron.NeutronClientWrapper):
+    """
+    Monitors reserved floating IPs which fail to clean up.
+    Deletes any floating IPs without an associated active reservation.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        return super(FloatingIpMonitorPlugin, cls).__new__(cls, *args, **kwargs)
+
+    def filter_allocations(self, reservation, resource_ids):
+        return [alloc for alloc
+                in reservation["floatingip_allocations"]
+                if alloc["floating_ip_id"] in resource_ids]
+
+    def get_reservations_by_resource_ids(self, resource_ids,
+                                         interval_begin, interval_end):
+        return db_utils.get_reservations_by_floatingip_ids(resource_ids,
+                                                          interval_begin,
+                                                          interval_end)
+
+    def get_unreservable_resourses(self):
+        return db_api.unreservable_fip_get_all_by_queries([])
+
+    def get_notification_event_types(self):
+        """Get event types of notification messages to handle."""
+        return ['service.update']
+
+    def notification_callback(self, event_type, payload):
+        return {}
+
+    def set_reservable(self, resource, is_reservable):
+        db_api.floatingip_update(resource["id"], {"reservable": is_reservable})
+        LOG.warn(
+            f"{resource['floating_ip_address']} "
+            f"{'recovered' if is_reservable else 'failed'} - setting reservable True"
+        )
+
+    def poll_resource_failures(self):
+        failed = []
+        recovered = []
+        dry_run = CONF[plugin.RESOURCE_TYPE].dry_polling_monitor
+        fips = db_api.floatingip_list()
+
+        def process_fip(fip):
+            fip_address = fip["floating_ip_address"]
+            fip_curr_reservation = db_utils.get_most_recent_reservation_info_by_fip_id(fip['id'])
+            if fip_curr_reservation and fip_curr_reservation['status'] == status.reservation.ACTIVE:
+                # This means that FIP works fine and no need to recover
+                LOG.debug(f"FIP {fip_address} is in active reservation {fip_curr_reservation['id']} - skipping")
+                return
+            fip_pool = neutron.FloatingIPPool(fip['floating_network_id'])
+            # check if the FIP is in neutron subnet allocation pools
+            try:
```

Comment:
This try block I am not sure about. It seems reasonable, but I don't fully understand all of the expected configuration. @msherman64 could you take a look?

Reply:
this try, except block looks good to me.
that exception indicates that the relevant ipv4 address is within a dhcp allocation pool on the public subnet.